Dockerfile

```diff
@@ -113,66 +145,72 @@ RUN apt-get update && apt-get install -y nodejs build-essential npm && \
     hash -d npm \
     npm install -g xml-js
 
-# Add moustache templates as mo
-RUN curl -sSO https://raw.githubusercontent.com/tests-always-included/mo/master/mo && \
-    chmod +x mo && \
-    mv mo /usr/local/bin
-
 # Clean up package manager
 RUN apt-get clean autoclean
 
 # Add random number generator to skip Docker building cache
 ADD http://www.random.org/strings/?num=10&len=8&digits=on&upperalpha=on&loweralpha=on&unique=on&format=plain&rnd=new /uuid
 
-# Elpis
+
+########################## ELPIS INSTALLATION ########################
+
 WORKDIR /
+
+# Elpis
 RUN git clone --depth=1 https://github.com/CoEDL/elpis.git
 
 # Elpis GUI
 RUN git clone --depth=1 https://github.com/CoEDL/elpis-gui.git
 
-# Example data
+WORKDIR /elpis-gui
+
+RUN npm install && \
+    npm run build
+
 WORKDIR /tmp
+
+# Example data
 RUN git clone --depth=1 https://github.com/CoEDL/toy-corpora.git
 
-# Set up zsh & ENV variables
+
+########################## ZSH CONFIG & ENV  ##########################
+
 WORKDIR /root
+
+# ZSH config
 RUN apt-get install zsh
 RUN chsh -s /usr/bin/zsh root
 RUN sh -c "$(wget -O- https://raw.githubusercontent.com/deluan/zsh-in-docker/master/zsh-in-docker.sh)" -- -t robbyrussell -p history-substring-search -p git
+
+# ENV
 RUN echo "export FLASK_ENV=development" >> ~/.zshrc
 RUN echo "export FLASK_APP=elpis" >> ~/.zshrc
 RUN echo "export LC_ALL=C.UTF-8" >> ~/.zshrc
 RUN echo "export LANG=C.UTF-8" >> ~/.zshhrc
 RUN echo "export PATH=$PATH:/venv/bin:/kaldi/src/bin/" >> ~/.zshrc
 RUN echo "alias run=\"flask run --host=0.0.0.0 --port=5000\"" >> ~/.zshrc
+RUN echo "alias run-win=\"flask run --host=127.0.0.1 --port=5000\"" >> ~/.zshrc
 RUN cat ~/.zshrc >> ~/.bashrc
-
-# Move ENV lines up. Putting here for now to build on top of cached builds
+# Make ENV vars available for non-interactive running
 ENV FLASK_ENV='development'
 ENV FLASK_APP='elpis'
 ENV LC_ALL=C.UTF-8
 ENV LANG=C.UTF-8
+# Don't do this, it breaks `python setup.py develop` below
```

Comment:
Was there an error message associated with this?

Reply:
Hmm.. all sorts of problems later, likely due to paths in the bashrc file. I didn't want to debug so much simpler to leave as is. I could remove the message but it could be a handy warning for later?

Reply:
In non-interactive `docker run`,  `.bashrc` isn't used, so the `echo export` vars are not available. Explicitly setting them with `ENV` works. I've updated it with comments on each block for clarity.